test_workflow.py

```diff
@@ -0,0 +1,58 @@
+import subprocess
```

Comment:
the github action should report how long it takes to run which will be more indicative of actual performance. see my caching prs i noticed caching would be slower than without